From: Sinthu Raja <sinthu.raja@ti.com>

Add node for the second instance of Cadence CSI2RX-1, DPHY-1, and TI's
CSI2RX wrapper. This enables CSI2 capture from the second instance of the
J721E camera subsystem.

Signed-off-by: Sinthu Raja <sinthu.raja@ti.com>
Reviewed-by: Pratyush Yadav <p.yadav@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721e-main.dtsi | 56 ++++++++++++++++++++++-
 1 file changed, 55 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
index 78f56841b13f..d8398eb14371 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
@@ -2345,7 +2345,7 @@ ti_csi2rx0: ticsi2rx@4500000 {
 		#size-cells = <2>;
 		ranges;
 
-		cdns_csi2rx: csi-bridge@4504000 {
+		cdns_csi2rx0: csi-bridge@4504000 {
 			compatible = "cdns,csi2rx";
 			reg = <0x0 0x4504000 0x0 0x1000>;
 			clocks = <&k3_clks 26 2>, <&k3_clks 26 0>, <&k3_clks 26 2>,
@@ -2382,10 +2382,64 @@ csi0_port4: port@4 {
 		};
 	};
 
+	ti_csi2rx1: ticsi2rx@4510000 {
+		compatible = "ti,j721e-csi2rx";
+		dmas = <&main_udmap 0x4960>;
+		dma-names = "rx0";
+		reg = <0x0 0x4510000 0x0 0x1000>;
+		power-domains = <&k3_pds 27 TI_SCI_PD_EXCLUSIVE>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		cdns_csi2rx1: csi-bridge@4514000 {
+			compatible = "cdns,csi2rx";
+			reg = <0x0 0x4514000 0x0 0x1000>;
+			clocks = <&k3_clks 27 2>, <&k3_clks 27 0>, <&k3_clks 27 2>,
+				<&k3_clks 27 2>, <&k3_clks 27 3>, <&k3_clks 27 3>;
+			clock-names = "sys_clk", "p_clk", "pixel_if0_clk",
+				"pixel_if1_clk", "pixel_if2_clk", "pixel_if3_clk";
+			phys = <&dphy1>;
+			phy-names = "dphy";
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				csi1_port0: port@0 {
+					reg = <0>;
+				};
+
+				csi1_port1: port@1 {
+					reg = <1>;
+				};
+
+				csi1_port2: port@2 {
+					reg = <2>;
+				};
+
+				csi1_port3: port@3 {
+					reg = <3>;
+				};
+
+				csi1_port4: port@4 {
+					reg = <4>;
+				};
+			};
+		};
+	};
+
 	dphy0: phy@4580000 {
 		compatible = "cdns,dphy";
 		reg = <0x0 0x4580000 0x0 0x1100>;
 		#phy-cells = <0>;
 		power-domains = <&k3_pds 147 TI_SCI_PD_EXCLUSIVE>;
 	};
+
+	dphy1: phy@4590000 {
+		compatible = "cdns,dphy";
+		reg = <0x0 0x4590000 0x0 0x1100>;
+		#phy-cells = <0>;
+		power-domains = <&k3_pds 148 TI_SCI_PD_EXCLUSIVE>;
+	};
 };
